devito/ir/support/guards.py

```diff
@@ -27,7 +27,12 @@ def canonical(self):
 
     @property
     def negated(self):
-        return negations[self.__class__](*self._args_rebuild)
+        try:
+            return negations[self.__class__](*self._args_rebuild)
+        except TypeError:
+            # Some complex relations involving custom SymPy subtypes (e.g.
+            # FieldFromPointer, IndexedPointer) mystically end up here
+            return negations[self.__class__](*self._args_rebuild, evaluate=False)
```

Comment:
it is implicit in the `try` part, no?

Reply:
I mean if you do the second one first you are more likely to "always" rebuild without evaluation while now you always evaluate unless it fails

Reply:
I'm not sure I follow you with this. If you pass `evaluate=False` it will never fail? are you suggesting to **avoid** evaluation, systematically? in that case, we could even drop the try-except